Mathlib/Order/InitialSeg.lean

```diff
@@ -462,34 +462,31 @@ theorem top_rel_top {r : α → α → Prop} {s : β → β → Prop} {t : γ 
 alias topLTTop := top_rel_top
 
 /-- Any element of a well order yields a principal segment. -/
--- The explicit typing is required in order for `simp` to work properly.
 @[simps!]
-def ofElement {α : Type*} (r : α → α → Prop) (a : α) :
-    @PrincipalSeg { b // r b a } α (Subrel r { b | r b a }) r :=
+def ofElement {α : Type*} (r : α → α → Prop) (a : α) : (Subrel r (r · a)) ≺i r :=
```

Comment:
```suggestion
def ofElement {α : Type*} (r : α → α → Prop) (a : α) : Subrel r (r · a) ≺i r :=
```
untested (but if it fails then it seems to me like the precedence of `≺i` has something weird going on)